bots/npm-update

```diff
@@ -104,18 +104,19 @@ def run(package, verbose=False, **kwargs):
 
     for package in packages:
 
-        # Don't try and upgrade http path specific dependencies
-        version = data["dependencies"][package]
-        if version[0].isalpha():
+        # Don't try and upgrade dependencies that are not purely numeric
+        if not before[package][0].isdigit():
             continue
 
         # Run npm upgrade for our package
         execute("npm", "upgrade", "--save", package)
 
         # Check if that did an upgrade
         update = package_json()
-        after = map_dict(update["dependencies"], lambda k, v: v.strip(" ^~=<>"))
-        if after != before:
+        after = update["dependencies"]
+        after[package] = after[package].strip(" ^~=<>")
```

Comment:
I actually wonder why that's necessary. If this already ensures that the original version starts with a digit (and not a ~ or ^), why would an npm update introduce this?

Reply:
The version package.json after `npm update` still has the "^" or "~" in front of it but we don't want that in the PR.  I am not sure about the rest of the characters, maybe `npm update` can introduce them as well?  But if `lstrip("^~")` is sufficient, it would indeed be better.

Reply:
Oh, right that. Yeah, your lstrip makes total sense.